src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -205,20 +206,47 @@ public static CommandLineBuilder CancelOnProcessTermination(this CommandLineBuil
 
         public static CommandLineBuilder UseDebugDirective(
             this CommandLineBuilder builder)
+            => builder.UseDebugDirective(TimeSpan.Zero);
+
+        public static CommandLineBuilder UseDebugDirective(
+            this CommandLineBuilder builder,
+            TimeSpan debuggerAttachTimeout)
         {
             builder.AddMiddleware(async (context, next) =>
             {
                 if (context.ParseResult.Directives.Contains("debug"))
                 {
-                    var process = Diagnostics.Process.GetCurrentProcess();
+                    const string environmentVariableName = "SYSTEM_COMMANDLINE_DEBUG_PROCESSES";
 
-                    var processId = process.Id;
+                    string debuggableProcessNames = GetEnvironmentVariable(environmentVariableName);
+                    if (string.IsNullOrWhiteSpace(debuggableProcessNames))
+                    {
+                        context.Console.Error.WriteLine("Debug directive specified, but no process names are listed as allowed for debug.");
+                        context.Console.Error.WriteLine($"Add your process name to the '{environmentVariableName}' environment variable.");
```

Comment:
This looks like it deserves a bit more detail, as the convention around multiple process names isn't evident.